### DISCUSSION TITLE
isnumeric (checking if string is numeric) Implementation?

### DISCUSSION THREAD

**Discussion by kat-rng** [micropython/micropython#11796] [Libraries & Drivers] [answered]
I am reading the output from a matrix keyboard library and am trying to check if the output (returned as a character from the library used) can be converted to a number. I saw that isalpha is implemented, but since the keypad can return "#" I am curious, **is there is any equivalent to isnumeric implemented?** And if not, is using a try except to get this behavior (as shown below) inadvisable? Thanks!
```{python}
def isnumeric(string):
    try: 
        float(string)
    except ValueError:
        return False
    return True
```

**Comment by 2dof** [comment]
You can check multiple instances  with  isinstance()  
```
value = '#'
if  isinstance(value,(float,int)): 
      print('is nmeric')
else:
     print('not nmeric')

```

**Reply by rkompass** [reply, marked as answer]
@kat-rng The try/except is appropriate. You have to consider that [issue](https://github.com/micropython/micropython/issues/11735).
So you could also check if the characters of the string are in '+-.eE0123456789'. This excludes something like '1_000e3' but that should be no problem.

**Reply by kat-rng** [reply]
Thank you. I was a little worried that running trys and excepts constantly may cause problems, but I take it that it's fine.

**Reply by jimmo** [reply]
@2dof this code does not actually do the thing that @kat-rng was asking for (namely to check if a string contains only the characters 0-9).

e.g.

```
>>> value = '#'
>>> isinstance(value,(float,int))
False
>>> value = '22'
>>> isinstance(value,(float,int))
False          <--- should be true
```

There are lots of ways to solve this particular problem... @kat-rng your solution with `float()` doesn't technically match CPython's implementation of str.isnumeric which is defined as "A string is numeric if all characters in the string are numeric and there is at least one character in the string."

Here is a simple version that doesn't involve try/except.

`return bool(value and not value.strip("0123456789"))`

**Comment by rkompass** [comment]
Thats one of many nice things: It becomes like if, else.--Diese Nachricht wurde von meinem Android Mobiltelefon mit GMX Mail gesendet.Am 16.06.23, 21:20 schrieb kat-rng ***@***.***>:

   
  Thank you. I was a little worried that running trys and accepts constantly may cause problems, but I take it that it's fine.
